### ISSUE TITLE
Unable to play media

### ISSUE THREAD

**Issue by jbalcao** [jarnedemeulemeester/findroid#255]
**Describe the bug**
Can't play any media after I restarted the app. Playing media after installation works but once the app is restarted, can no longer play.

**To Reproduce**
1. I've installed the app from github release.
2. After initial use, playing media works.
3. Restarted the app.
4. Playing media no longer works. Tried enabling and disabling mpv player, still not working.

**Expected behavior**
Able to play movies/shows

**Screenshots**
Player just stucks like this
![Screenshot_20230131-155448.png](https://user-images.githubusercontent.com/88887245/215699801-567ee4f3-9551-48f8-86c1-c138c41e5168.png)

**Device info (please complete the following information):**
 - Device:  Google Pixel 7
 - Android version: 13
 - App version: V.0.10
 - Jellyfin version: 10.8.9

**Comment by jbalcao** [comment]
Yes, clearing the app data works again. But now, enabling the mpv player will not work.

**Comment by jarnedemeulemeester** [comment]
Okay so now you can still play media after restarting the app?
Is the mpv problem the same as your original report (the media playback is never started, infinite loading)?

What type of media files are you trying to play? Container format, codecs, etc.. Any information about the files is helpful :)

**Comment by jbalcao** [comment]
Yes, the mpv player problem is the same. Just infinite loading. My media formats are mp4 and mkv. H264 codec.

**Comment by TnZzZHlp** [comment]
You can try to change the 'Hardware decoding' option to mediacodec or no , and try to play the video again
The "Hardware decoding" option is in Settings - Player

**Comment by jbalcao** [comment]
I also tried changing settings under the mpv player. Nothing works. 🙁 I'll just disable mpv player for now.

**Comment by SMSourov** [comment]
Same issue here too. In my case, the folder that is in Music Videos category doesn't play at all. But others are playing fine. 😐

**Comment by tam1m** [comment]
FFmpeg seems to have an issue with the server certificates. The certs are are fine though. 

This only happens when using mpv.



```
V/mpv: [ffmpeg:error] tls: A fatal alert message was received from the peer, has the peer a correct certificate?
V/mpv: [stream:error] Failed to open https://xxxx.xxxxxx.xx/Videos/251aa3ca-1138-456e-af91-542dc46b8d82/stream?static=true&mediaSourceId=251aa3ca1138456eaf91542dc46b8d82&streamOptions=%7B%7D.
V/mpv: event: end-file
V/mpv: event: start-file
V/mpv: [cplayer:v] Opening failed or was aborted: https://xxxx.xxxxxx.xx/Videos/251aa3ca-1138-456e-af91-542dc46b8d82/stream?static=true&mediaSourceId=251aa3ca1138456eaf91542dc46b8d82&streamOptions=%7B%7D
V/mpv: [cplayer:v] finished playback, loading failed (reason 4)
V/mpv: [cplayer:info] 
```

[Full log of this happening](https://github.com/jarnedemeulemeester/findroid/files/10608403/findroid_mpv.log)

**Comment by jbalcao** [comment]
Yeah, confirmed working via http/ip.

**Comment by tam1m** [comment]
So the issue seems to be tls 1.3. Upstream shinchiro/mpv-winbuild-cmake@358b4a3 moved from libressl to mbedtls, which seems to have some kind of problem with TLS 1.3.

So until this gets sorted out the only workaround for now is to allow tls 1.2 with your reverseproxy.
For nginx `ssl_protocols TLSv1.2 TLSv1.3;` would to that.

For reference JMP has the same issue [issue](jellyfin/jellyfin-media-player/issues/375)

**Comment by jarnedemeulemeester** [comment]
@jbalcao  @tam1m  Thanks for the research.

Findroid uses it's own custom build of [mpv](https://github.com/jarnedemeulemeester/libmpv-android) which uses Mbed TLS 2.28.1 which has no support for TLS 1.3.
I will try to update to Mbed TLS 3.3.0 which should support TLS 1.3.

Once I have upgraded Mbed TLS I will force my server to only accept TLS 1.3 connections and report back with my findings.

**Comment by jarnedemeulemeester** [comment]
Okay so building with Mbed TLS 3.3.0 and enabling TLS 1.3 does still seems to have some issues.
```
V/mpv: [ffmpeg:error] tls: mbedtls_ssl_handshake returned -0x6c00
```

If I build Mbed TLS for my pc and run the `ssl_client2` program it works fine.

Also it seems that the TLS 1.3 implementation in Mbed TLS is still incomplete. https://github.com/Mbed-TLS/mbedtls/blob/development/docs/architecture/tls13-support.md

There is also the active development repo for TLS 1.3 but I didn't test that one since I would like to stick to the official release. https://github.com/hannestschofenig/mbedtls

**Comment by jarnedemeulemeester** [comment]
Has been fixed in v0.15.3
